soc/arm/st_stm32/common/Kconfig.defconfig.series

```diff
@@ -71,8 +71,4 @@ config DMA_STM32
 
 endif # DMA
 
-config MEMC_STM32
-	default y
-	depends on MEMC
```

Comment:
All other peripherals do the same (see the lines above). This enables MEMC_STM32 automatically when MEMC is enabled, so I think it does its job.

Reply:
Indeed, you should keep this line and remove `CONFIG_MEMC_STM32=y` in `tests/drivers/memc/stm32_sdram/prj.conf `

Reply:
Don't think the other peripheral settings are right either.

So with that block in Kconfig.defconfig.series, MEMC can be selected for every STM32 part (current code) I can, say, select an F103 board, enable MEMC, enable MEMC_STM32 and get a config that does not build cause there's no fmc node.

With that code removed, MEMC_STM32 can only be selected if MEMC is selected (the Kconfig is not included if not) and the node is present (does not build otherwise anyway),

If we want to get rid of the line in prj.conf, I can switch the MEMC (which depends on the node) to default y, in that case it can only be selected if the node is defined and it also does not have to be selected explicitly, and will just go in if MEMC is enabled. I'll do that change, lmk what you think.

Reply:
The other options should be reviewed as well, for example I can do

`west build -p -b nucleo_wl55jc samples/basic/blinky`

and then in menuconfig I can enable `USB` and `USB_DC_STM32` gets enabled automatically (the device does not have an USB controller), resulting in a broken build.

Reply:
ok, right with the compat enabled, this should indeed be removed. Txs